tests/testTransforms.py

```diff
@@ -0,0 +1,468 @@
+#!/usr/bin/env python
+from __future__ import absolute_import, division, print_function
+
+#
+# LSST Data Management System
+# Copyright 2016 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+import unittest
+import numpy
+import lsst.utils.tests
+import lsst.pex.exceptions
+import lsst.afw.geom
+from lsst.meas.astrom import (
+    PolynomialTransform,
+    ScaledPolynomialTransform,
+    SipForwardTransform,
+    SipReverseTransform,
+    ScaledPolynomialTransformFitter
+)
+
+def makeRandomCoefficientMatrix(n):
+    matrix = numpy.random.randn(n, n)
+    for i in xrange(1, n):
+        matrix[i, (n-i):] = 0;
+    return matrix
+
+def makeRandomAffineTransform():
+    return lsst.afw.geom.AffineTransform(
+        lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+        lsst.afw.geom.Extent2D(*numpy.random.randn(2))
+    )
+
+def makeRandomPolynomialTransform(order, sip=False):
+    xc = makeRandomCoefficientMatrix(order + 1)
+    yc = makeRandomCoefficientMatrix(order + 1)
+    if sip:
+        xc[0,0] = 0
+        yc[0,0] = 0
+        xc[0,1] = 0
+        yc[0,1] = 0
+        xc[1,0] = 0
+        yc[1,0] = 0
+    return PolynomialTransform(xc, yc)
+
+def makeRandomScaledPolynomialTransform(order):
+    return ScaledPolynomialTransform(
+        makeRandomPolynomialTransform(order),
+        makeRandomAffineTransform(),
+        makeRandomAffineTransform()
+    )
+
+def makeRandomSipForwardTransform(order):
+    return SipForwardTransform(
+        lsst.afw.geom.Point2D(*numpy.random.randn(2)),
+        lsst.afw.geom.LinearTransform(numpy.random.randn(2,2)),
+        makeRandomPolynomialTransform(order, sip=True)
+    )
+
+def makeRandomSipReverseTransform(order):
+    origin = lsst.afw.geom.Point2D(*numpy.random.randn(2))
+    cd = lsst.afw.geom.LinearTransform(numpy.random.randn(2,2))
+    poly = makeRandomPolynomialTransform(order, sip=False)
+    return SipReverseTransform(origin, cd, poly)
+
+
+class TransformTestMixin(object):
+
+    def checkTransformEquivalence(self, a, b, rtol=1E-8):
```

Comment:
I've renamed it as you've suggested.

I don't think I'm going to move it somewhere more general right now, though, because I think the domain/range questions are very important for testing transforms more generally, and I don't want to put in the work right now to make this utility function handle more general domains and ranges.